portafly/src/components/AppLayout.tsx

```diff
@@ -0,0 +1,172 @@
+  children,
+}) => {
+  const [isNavOpen, setIsNavOpen] = React.useState(startWithOpenNav)
+  const [isMobileView, setIsMobileView] = React.useState(true)
+  const [isNavOpenMobile, setIsNavOpenMobile] = React.useState(false)
+  const [breadcrumb, setBreadcrumb] = React.useState<React.ReactNode | undefined>()
+  const previousBreadcrumb = React.useRef<React.ReactNode | null>()
+
+  const handleSetBreadcrumb = React.useCallback(
+    (newBreadcrumb: React.ReactNode) => {
+      if (previousBreadcrumb.current !== newBreadcrumb) {
+        previousBreadcrumb.current = newBreadcrumb
+        setBreadcrumb(previousBreadcrumb.current)
+      }
+    },
+    [setBreadcrumb, previousBreadcrumb]
+  )
+
+  const onNavToggleMobile = React.useCallback(() => {
+    setIsNavOpenMobile(!isNavOpenMobile)
+  }, [setIsNavOpenMobile, isNavOpenMobile])
+
+  const onNavToggle = React.useCallback(() => {
+    setIsNavOpen(!isNavOpen)
+  }, [setIsNavOpen, isNavOpen])
+
+  const onPageResize = (props: { mobileView: boolean; windowSize: number }) => {
+    setIsMobileView(props.mobileView)
+  }
+
+  React.useEffect(() => {
+    setIsNavOpen(startWithOpenNav)
+  }, [startWithOpenNav, setIsNavOpen])
+
+  const isVertical = navVariant === 'vertical'
+  const variant = isVertical ? NavVariants.default : NavVariants.horizontal
+
+  const Navigation = React.useMemo(
+    () =>
+      navItems.length > 0 ? (
+        <Nav id="nav-primary-simple" theme={theme}>
+          <NavList id="nav-list-simple" variant={variant}>
+            {navItems.map((navItem, idx) => {
+              if (navItem && navItem.hasOwnProperty('items') && isVertical) {
+                return navGroupsStyle === 'expandable' ? (
+                  <AppNavExpandable
+                    {...(navItem as IAppNavExpandableProps)}
+                    key={idx}
+                  />
+                ) : (
+                  <AppNavGroup {...(navItem as IAppNavGroupProps)} key={idx}/>
+                )
+              } else {
+                return (
+                  <AppNavItem {...(navItem as IAppNavItemProps)} key={idx}/>
+                )
+              }
+            })}
+          </NavList>
+        </Nav>
+      ) : null,
+    [isVertical, navGroupsStyle, navItems, theme, variant]
+  )
+
+  const Header = React.useMemo(
+    () => (
+      <PageHeader
+        logo={logo}
+        logoProps={logoProps}
+        avatar={avatar}
+        toolbar={toolbar}
+        showNavToggle={isVertical}
+        isNavOpen={isVertical ? isNavOpen : undefined}
+        onNavToggle={isMobileView ? onNavToggleMobile : onNavToggle}
+        topNav={isVertical ? undefined : Navigation}
+      />
+    ),
+    [
+      logo,
+      logoProps,
+      avatar,
+      toolbar,
+      isVertical,
+      isNavOpen,
+      isMobileView,
+      onNavToggle,
+      onNavToggleMobile,
+      Navigation,
+    ]
+  )
+
+  const Sidebar = React.useMemo(
+    () =>
+      navVariant === 'vertical' ? (
```

Comment:
Instead of returning undefined:
```suggestion
      navVariant === 'vertical' && (
```